### PR TITLE
Encryption code cleanup to isolate provider

### PR DESCRIPTION
**Pull request by bhariharan**
Some `BouncyCastle` functionality is being removed in `Android P`. Refactored the code to help us pivot from `BouncyCastle` and use `Conscrypt` instead if we want in the future.

### CODE REVIEW

File: libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/security/Encryptor.java [hunk 1]
```diff
@@ -58,67 +55,8 @@
...
-    private static String bestCipherAvailable;
-    private static boolean isFileSystemEncrypted;
-
-    /**
-     * Initializes this module.
-     *
-     * @param ctx Context.
-     * @return True - if the cryptographic module was successfully initialized, False - otherwise.
-     */
-    public static boolean init(Context ctx) {
```
Comment (bhariharan) [inline libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/security/Encryptor.java]: Most of this code was legacy code that's not required anymore for 2 reasons:
1. File system encryption is available by default even with our `minApi`, i.e. `API 21`.
2. The reason we had hardcoded `BouncyCastle` in here was that it wasn't the default or even available on some devices prior to `Eclair` or `Froyo` (`Android 2.x`), which we haven't supported in years now.

File: libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/security/Encryptor.java [hunk 2]
```diff
@@ -58,67 +55,8 @@
...
-        try {
-            Mac.getInstance(MAC_TRANSFORMATION, "BC");
-        } catch (GeneralSecurityException e) {
-            SalesforceAnalyticsLogger.e(ctx, TAG, "No MAC transformation available", e);
-            return false;
-        }
-        return true;
-    }
-
-    private static Cipher getBestCipher() throws GeneralSecurityException {
```
Comment (bhariharan) [inline libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/security/Encryptor.java]: Same reasons as listed above. The best cipher is the one provided by the OS now, i.e. `Conscrypt` on `Android P` and higher.

File: libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/security/Encryptor.java [hunk 3]
```diff
@@ -58,67 +55,8 @@
...
-        }
-        return cipher;
-    }
-
-    /**
-     * Checks if the file system is encrypted.
-     *
-     * @return True - if file system encryption is available and active, False - otherwise.
-     */
-    public static boolean isFileSystemEncrypted() {
```
Comment (bhariharan) [inline libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/security/Encryptor.java]: This is always going to return `true` and it's not used anywhere or have any value anymore.

File: libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/security/Encryptor.java [hunk 4]
```diff
@@ -302,8 +240,8 @@ public static String hash(String data, String key) {
             // Signs with SHA-256.
             byte [] keyBytes = key.getBytes(UTF8);
             byte [] dataBytes = data.getBytes(UTF8);
-            Mac sha = Mac.getInstance(MAC_TRANSFORMATION, "BC");
```
Comment (bhariharan) [inline libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/security/Encryptor.java]: Use the default from the OS now, it's likely better than `BouncyCastle`.

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/app/SalesforceSDKManager.java [hunk 5]
```diff
@@ -454,9 +454,6 @@ public static void initInternal(Context context) {
         // Upgrades to the latest version.
         SalesforceSDKUpgradeManager.getInstance().upgrade();
 
-        // Initializes the encryption module.
-        Encryptor.init(context);
```
Comment (bhariharan) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/app/SalesforceSDKManager.java]: This made sense in the `Eclair` and `Froyo` days, but I don't think it's required anymore.

**Comment by wmathurin** [comment]
Is some kind of migration needed when moving from 6.1 (or older) to 6.2?

File: libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/security/Encryptor.java [hunk 6]
```diff
@@ -451,7 +390,21 @@ public static String decryptBytes(byte[] data, byte[] key, byte[] iv) {
...
+            cipher = Cipher.getInstance(PREFER_CIPHER_TRANSFORMATION, getEncryptionProvider());
+        } catch (Exception e) {
+            SalesforceAnalyticsLogger.e(null, TAG,
+                    "No cipher transformation available", e);
+        }
+        return cipher;
+    }
+
+    private static String getEncryptionProvider() {
+        return BOUNCY_CASTLE;
```
Comment (bhariharan) [inline libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/security/Encryptor.java]: Added `Bouncy Castle` as the default back in here.

**Comment by bhariharan** [comment]
Tested upgrade scenarios with passcode. Works fine.

**Pull request merged**